### PR TITLE
PipeConsensus: fix concurrent problems regarding to initiate receiver and progressIndex & optimize receiver logs & fix npe

### PR DESCRIPTION
**Pull request by Pengzna**
As title:
1. fix concurrent problems regarding to initiate receiver
2. fix concurrent problems regarding to progressIndex
3. fix update bugs in updating progressIndex
4. fix npe when initiate syncConnector
5. optimize receiver's logs when create dirs.

This pr will fix TIMECHO-771

### CODE REVIEW

**Comment by Pengzna** [comment]
@OneSizeFitsQuorum @Caideyipi PTAL

**Review by Caideyipi — APPROVED** [review]
LGTM

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/consensus/ProgressIndexDataNodeManager.java [hunk 1]
```diff
@@ -74,9 +74,13 @@ private void recoverMaxProgressIndexFromDataRegion() {
...
                         extractLocalSimpleProgressIndex(progressIndex));
               }
-              groupId2MaxProgressIndex
-                  .computeIfAbsent(dataRegionId, o -> MinimumProgressIndex.INSTANCE)
-                  .updateToMinimumEqualOrIsAfterProgressIndex(maxProgressIndex);
+              ProgressIndex correspondingProgressIndex =
+                  groupId2MaxProgressIndex
+                      .computeIfAbsent(dataRegionId, o -> MinimumProgressIndex.INSTANCE)
+                      .updateToMinimumEqualOrIsAfterProgressIndex(maxProgressIndex);
+              // Need to update map by hand due to the syntax of
```
[the remark this answers is not in the document]
Comment (Pengzna) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/consensus/ProgressIndexDataNodeManager.java]: ok

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**